src/ring_theory/local_properties.lean

```diff
@@ -0,0 +1,353 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import ring_theory.localization
+import data.equiv.transfer_instance
+import group_theory.submonoid.pointwise
+import ring_theory.nilpotent
+
+/-!
+# Local properties of commutative rings
+
+In this file, we provide the proofs of various local properties.
+
+## Naming Conventions
+
+* `localization_P` : `P` holds for `S⁻¹R` if `P` holds for `R`.
+* `P_of_localization_maximal` : `P` holds for `R` if `P` holds for `Aₘ` for all maximal `m`.
+* `P_of_localization_span` : `P` holds for `R` if given a spanning set `{fᵢ}`, `P` holds for all
+  `A_{fᵢ}`.
```

Comment:
Does it make sense to turn these into actual definitions. So that you can write something along the lines of `is_local @finite` or `is_local_maximal @finite` etc...
To be precise: can we turn "the property `P` is local" into code?